presto-iceberg/README.md

```diff
@@ -0,0 +1,112 @@
+This plugin allows Presto to interact with [Iceberg](https://iceberg.apache.org/) tables.
+
+## Status
+
+Currently, this plugin supports create, CTAS, drop, rename, and reading from Iceberg tables.
+It also supports adding, dropping, and renaming columns.
+
+## Configuration
+
+Iceberg supports the same metastore configuration properties as the Hive connector.
+At a minimum, `hive.metastore.uri` must be configured:
+
+```
+connector.name=iceberg
+hive.metastore.uri=thrift://localhost:9083
+```
+
+## Unpartitioned Tables
+
+``` sql
+CREATE TABLE iceberg.testdb.sample (
+    i int, 
+    s varchar
+)
+```
+
+## Partitioned Tables
+
+Iceberg supports partitioning by specifying transforms over the table columns.
+A partition is created for each unique tuple value produced by the transforms.
+Identity transforms are simply the column name. Other transforms:
+
+* `year(ts)`
+* `month(ts)`
+* `day(ts)`
+* `hour(ts)`
+* `bucket(x, 512)`
+* `truncate(s, 16)`
+
+``` sql
+CREATE TABLE iceberg.testdb.sample_partitioned (
+    b boolean,
+    dateint integer,
+    l bigint,
+    f real,
+    d double,
+    de decimal(12,2),
+    dt date,
+    ts timestamp,
+    s varchar,
+    bi varbinary
+ )
+WITH (partitioning = ARRAY['dateint', 's'])
+```
+
+## Migrating existing tables
+
+The connector can read from or write to Hive tables that have been migrated to Iceberg.
+Currently, there is no Presto support to migrate Hive tables to Presto, so you will
+need to use either the Icerberg API or Spark.
+
+## System Tables and Columns
+
+The connector supports `$partitions` as a substitute for Hive's `SHOW PARTITIONS`.
+The differences are that it returns some partition metrics for each partition value
+and you can also use it for unpartitioned tables.
+
+Iceberg supports `$snapshot_id` and `$snapshot_timestamp_ms` as hidden columns.
+These columns allow users to query an old version of the table. Think of this
+as a time travel feature which lets you query your table's snapshot at a given time.
+
+## TODO
```

Comment:
We also want to add support for ORC and Avro? Can list them as TODOs here.

Reply:
Those don’t seem to be fully specified or implemented in the main Iceberg project yet. I’d like to do ORC since Presto has a native writer but stopped when I saw that.

This TODO list is intended to capture items required to announce the connector as ready to use. As part of that, we’ll delete this README and replace it with real documentation. We should file issues to track such future enhancements.

Reply:
@electrum 
Iceberg repo's README says that
> iceberg-orc is an optional module for working with tables backed by ORC files (experimental)`

But I think it may be okay to add ORC support now. Iceberg's spec does involve ORC-related requirements. (https://iceberg.apache.org/spec/#orc). And Spark's ORC support has been introduced in https://github.com/apache/incubator-iceberg/pull/139.

cc: @rdblue Any comment?

Reply:
ORC is not fully implemented yet, so I don't recommend adding support. Specifically, tracking columns by ID is not yet complete so it isn't possible to project columns correctly yet.

Reply:
@rdblue thanks for the confirmation. The comment about column IDs in the specification was worrying. There are also no tests in the Iceberg ORC module, so I assumed it wasn't complete yet. (`TestOrcWrite` in the Spark module is also quite sparse)

Reply:
Yeah, part of getting out of "experimental" status is finishing the spec for ORC.